app/services/activitypub/fetch_replies_service.rb

```diff
@@ -8,9 +8,13 @@ class ActivityPub::FetchRepliesService < BaseService
 
   def call(reference_uri, collection_or_uri, max_pages: 1, allow_synchronous_requests: true, request_id: nil)
     @reference_uri = reference_uri
-    @allow_synchronous_requests = allow_synchronous_requests
+    return if !allow_synchronous_requests & !collection_or_uri.is_a?(Hash)
 
-    @items, n_pages = collection_items(collection_or_uri, max_pages: max_pages)
+    # if given a prefetched collection while forbidding synchronous requests,
+    # process it and return without fetching additional pages
+    max_pages = 1 if !allow_synchronous_requests & collection_or_uri.is_a?(Hash)
```

Comment:
I believe this replicates the previous behavior, where if one passed `allow_synchronous_requests: False` but provided the prefetched collection that it would process it and not try and fetch any further pages. sorry if this is an awkward way to do it, i still don't exactly know the idiomatic way to do this in ruby.